micrometer-docs-generator-commons/src/main/java/io/micrometer/docs/commons/ParsingUtils.java

```diff
@@ -131,12 +132,27 @@ private static String tryToReadNameFromConventionClass(Path file, String classNa
                 if (interfaces.stream().noneMatch(s -> s.contains(ObservationConvention.class.getSimpleName()))) {
                     return null;
                 }
-                MethodSource<JavaClassSource> name = ((JavaClassImpl) actualConventionImplementation).getMethod("getName");
+                MethodSource<?> name = ((JavaClassImpl) actualConventionImplementation).getMethod("getName");
+                if (name == null) {
+                    // look for the implementing interfaces
+                    for (String iface : interfaces) {
```

Comment:
Would this cover level > 2?

Reply:
not with this impl unfortunately...

Reply:
It is definitely better than it was before this PR - ty.